From d02340d90c9d30d44c962bea7171db3fe3bfba8e Mon Sep 17 00:00:00 2001
From: Alexandre Oliva <oliva@gnu.org>
Date: Wed, 6 Feb 2013 15:27:13 -0200
Subject: [PATCH] silence logrotate some more
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

I was getting email with logrotate error output from “which invoke-rc.d”
on systems without an invoke-rc.d.  This patch silences it.

Silence stderr from which when running logrotate

From: Alexandre Oliva <oliva@gnu.org>

Signed-off-by: Alexandre Oliva <oliva@gnu.org>
---
 src/logrotate.conf |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/logrotate.conf b/src/logrotate.conf
index 62101b3..e49285a 100644
--- a/src/logrotate.conf
+++ b/src/logrotate.conf
@@ -4,13 +4,13 @@
     compress
     sharedscripts
     postrotate
-        if which invoke-rc.d > /dev/null && [ -x `which invoke-rc.d` ]; then
+        if which invoke-rc.d > /dev/null 2>&1 && [ -x `which invoke-rc.d` ]; then
             invoke-rc.d ceph reload >/dev/null
-        elif which service > /dev/null && [ -x `which service` ]; then
+        elif which service > /dev/null 2>&1 && [ -x `which service` ]; then
             service ceph reload >/dev/null
         fi
         # Possibly reload twice, but depending on ceph.conf the reload above may be a no-op
-        if which initctl > /dev/null && [ -x `which initctl` ]; then
+        if which initctl > /dev/null 2>&1 && [ -x `which initctl` ]; then
             # upstart reload isn't very helpful here:
             #   https://bugs.launchpad.net/upstart/+bug/1012938
             for type in mon osd mds; do
